src/NewPost.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { data } from "./data";
+
+export class NewPost extends Component {
+  state = {
+    id: 0,
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: ""
+  };
+
+  createDateOnSubmit = () => {
+    const date = new Date();
+    let year = date.getFullYear();
+    let month = date.getMonth();
+    let day = date.getDate();
+    const dateNow = `${year}-${month}-${day}`;
+    return dateNow;
+  };
+
+  createUniqueIdOnSubmit = () => {
+    const id = data.length;
+    return id;
+  };
+
+  convertTagsToArray = tags => {
+    let resultArray = [];
+
+    if (tags != null) {
```

Comment:
why is this `if` check necessary? have you seen a situation where tags was null?

Reply:
my misunderstanding. null means 'no value' not 'empty'